apiserver/common/credentialcommon/cloudcredential.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package credentialcommon
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/state"
+)
+
+// CloudCredentialInterface is an interface for manipulating cloud credential.
+type CloudCredentialInterface interface {
```

Comment:
We don't normally call interfaces `SomethingInterface` - `CredentialManager` maybe? Other APIs often just call this `Backend`, that would be ok too.

Reply:
There are a few places in apiserver/common package that do use *Interface as a naming convention for this particular type of a construct... 
I am not too attached to the name though as it's the functionality that I am after, so will rename to Backend :D